app/models/conversion_host.rb

```diff
@@ -274,16 +283,46 @@ def miq_ssh_util_args_manageiq_providers_openstack_cloudmanager_vm
   # +extra_vars+ option should be a hash of key/value pairs which, if present,
   # will be passed to the '-e' flag.
   #
-  def ansible_playbook(playbook, extra_vars = {})
-    command = "ansible-playbook #{playbook} -i #{ipaddress}"
+  def ansible_playbook(playbook, extra_vars = {}, auth_type = nil)
+    task = MiqTask.all.select { |t| t.context_data.present? && t.context_data[:conversion_host_id] == id }.sort_by(&:created_on).last
```

Comment:
This is where having proper associations would help, this is bringing back _every_ miq_task and loading up the context_data hash in ruby :scream:

Reply:
We could use a `.where` clause here instead couldn't we? Totally untested, but this is the general idea:

```
MiqTask.where.not(:context_data => nil)
       .where(:context_data[:conversion_host_id]) => id)
```

Reply:
I'd rather pass the task_id to the calling method and update the task there so we don't have to do any searching at all

Reply:
Indeed, we have the task id, so I changed the code to pass it to the `*_conversion_host_role` methods, that in turn pass it to `ansible_playbook`.